massa-pos-exports/src/cycle_info.rs

```diff
@@ -33,6 +99,100 @@ pub struct CycleInfo {
     pub rng_seed: BitVec<u8>,
     /// Per-address production statistics
     pub production_stats: PreHashMap<Address, ProductionStats>,
+    /// Hash of the self cycle state
+    pub hash: Hash,
+}
+
+impl CycleInfo {
+    /// Create a new `CycleInfo` and compute its hash
+    pub fn new_with_hash(
+        cycle: u64,
+        complete: bool,
+        roll_counts: BTreeMap<Address, u64>,
+        rng_seed: BitVec<u8>,
+        production_stats: PreHashMap<Address, ProductionStats>,
+    ) -> Self {
+        let hash_computer = CycleInfoHashComputer::new();
+        let mut hash = Hash::from_bytes(CYCLE_INFO_HASH_INITIAL_BYTES);
+
+        // compute the cycle hash
+        hash ^= hash_computer.compute_cycle_hash(cycle);
+        hash ^= hash_computer.compute_complete_hash(complete);
```

Comment:
Because you can't just compute one hash from the serialized cycle on new, it would be different than a cycle that was partially updated multiple times on a running node. You are right about the collide though, added an identifier to avoid that.

Reply:
Shouldn't you add identifier to all of them ? Collision can't happens with seed or rolls ?

Reply:
There is a very slight chance that bitvec might collide with something indeed, will add an ident for it too

Reply:
Collision on this is low enough for us to not care about it

Reply:
I did some other updates on the CycleInfo hash computation, would be great if you can take a look again at `cycle_info.rs`